File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int size = Integer.parseInt(br.readLine());
        Queue<Integer> queue = new LinkedList<>();
        StringBuilder sb = new StringBuilder();
        int last = 0;
        for (int i = 0; i < size; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String check = st.nextToken();
            if (check.equals("push")) {
                last = Integer.parseInt(st.nextToken());
                queue.add(last);
                continue;
            }

            switch (check) {
                case "pop":
                    if (queue.isEmpty()) sb.append(-1).append("\n");
                    else sb.append(queue.poll()).append("\n");
                    break;

                case "size":
                    sb.append(queue.size()).append("\n");
                    break;

                case "empty":
                    if (queue.isEmpty()) sb.append(1).append("\n");
                    else sb.append(0).append("\n");
                    break;

                case "front":
                    if (queue.isEmpty()) sb.append(-1).append("\n");
                    else sb.append(queue.peek()).append("\n");
                    break;

                case "back" :
                    if (queue.isEmpty()) sb.append(-1).append("\n");
                    else sb.append(last).append("\n");
                    break;

            }
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/18258. 큐 2/큐 2.py
from collections import deque
import sys

size = int(sys.stdin.readline())
queue = deque([])
ans_list = []
for i in range(size):
    check = list(map(str, sys.stdin.readline().split()))

    if len(check) != 1:
        queue.append(int(check[1]))

    else:
        if check[0] == 'pop':
            if queue:
                ans_list.append(queue.popleft())

            else:
                ans_list.append(-1)

        elif check[0] == 'size':
            ans_list.append(len(queue))

        elif check[0] == 'empty':
            if queue:
                ans_list.append(0)

            else:
                ans_list.append(1)

        elif check[0] == 'front':
            if queue:
                ans_list.append(queue[0])

            else:
                ans_list.append(-1)

        elif check[0] == "back":
            if queue:
                ans_list.append(queue[len(queue) - 1])

            else:
                ans_list.append(-1)

for i in ans_list:
    sys.stdout.write(str(i) + "\n")